Tensile/KernelWriterAssembly.py

```diff
@@ -1568,6 +1571,32 @@ def defineMACMacro(self, kernel, innerUnroll):
             else:
               printExit("Half-precision not supported for arch=%u" % self.version )
 
+      # integer i8
+      elif kernel["ProblemType"]["DataType"].isInt8x4():
+        for b in range(0, kernel["ThreadTile1"]):
+          for a in range(0, kernel["ThreadTile0"]):
+            if self.version == (8,0,3):
+              kStr += self.comment3("int8 not implemented yet for gfx803:")
+            elif self.version == (9,0,0):
+              kStr += self.comment3("int8 not implemented yet for gfx900:")
+            elif self.version == (9,0,6):
+              for iui in range(0, innerUnroll):
+                cidx = a + b*kernel["ThreadTile0"] + 0
+                cStr = "v[%s+%u+%u*%u]" % ("vgprValuC", a, b, kernel["ThreadTile0"])
+                aStr = "v[%s+%u]"       % ("vgprValuA_X%u_I%u"%(m,iui), a)
+                bStr = "v[%s+%u]"       % ("vgprValuB_X%u_I%u"%(m,iui), b)
+# this will be same as "single precision" except for the dot4 instruction replacing the v_mac_f32 instruction
+#               kStr += "v_mac_f32 %s, %s, %s%s" % (cStr, aStr, bStr, self.endLine) # single precision
+                kStr += "v_dot4_i32_i8  %s, %s, %s, %s op_sel:[0,0] op_sel_hi:[1,1] //valuC[%u]%s" % (cStr, aStr, bStr, cStr, cidx, self.endLine)
+
+                if macIdx == kernel["PerformanceWaitLocation"]:
```

Comment:
I suspect we don't need either of these waits anymore

Reply:
I removed the wait, and tests pass, this is limited testing. Can the s_barrier also be removed?

Reply:
Also removed s_barrier